apps/api/src/report/report.service.ts

```diff
@@ -269,46 +272,120 @@ export class ReportService {
 
   /**
    * Report 9
+   * Get average time spent on the stages, NNAS, BCCNM & NCAS, Recruitment by Health Authorities, and Immigration
+   * including overall time from the start to immigration completion and average time to hire.
+   *
+   * NOTE: Counting only hired applicants completed immigration
+   *
    * @param t end date of report, YYYY-MM-DD
    */
   async getAverageTimeWithEachStakeholderGroup(statuses: Record<string, string>, t: string) {
     const { to } = this.captureFromTo('', t);
     this.logger.log(`getAverageTimeWithEachStakeholderGroup: apply filter till ${to} date)`);
 
-    const entityManager = getManager();
-    const data = await entityManager.query(
-      this.reportUtilService.averageTimeWithEachStakeholderGroupQuery(statuses, to),
-    );
-    this.logger.log(
-      `getAverageTimeWithEachStakeholderGroup: query completed a total of ${data.length} record returns`,
```

Comment:
any chance of keeping these logs in report 9 and 10?

Reply:
Because we can view the list of applicants with relevant milestones by querying on the view, I don't think logging of intermediate queries is required.